packages/as-proto-gen/src/generate/message.ts

```diff
@@ -71,8 +62,28 @@ export function generateMessage(
     `
     : "";
 
+  const MessageClass = `
+    ${canMessageByUnmanaged(messageDescriptor, fileContext) ? "@unmanaged" : ""}
+    export class ${Message} {
+      ${generateEncodeMethod(messageDescriptor, fileContext)}
+      ${generateDecodeMethod(messageDescriptor, fileContext)}
+
+      ${generateMessageFieldsDeclarations(messageDescriptor, fileContext)}
+
+      ${generateMessageConstructor(messageDescriptor, fileContext)}
+    }
+  `;
+
+  fileContext.registerImport("Protobuf", "as-proto");
```

Comment:
Adding a utility function to decode message.

Reply:
to be fully correct, I think you should also add after `const Message = fileContext.registerDefinition(messageName, messageNamespace);` line:
```typescript
fileContext.registerDefinition(`decode${Message}`);
```
this ensures that there are no name conflicts in the file scope